deploy.php

```diff
@@ -12,32 +12,42 @@
 set('repository', '$repository');
 set('production_url', '$productionUrl');
 set('production_user', '$productionUser');
+set('php_version', '8.0');
 
 // Define staging
 host('dev02.sumocoders.eu')
     ->setRemoteUser('sites')
     ->set('labels', ['stage' => 'staging'])
     ->set('deploy_path', '~/apps/{{client}}/{{project}}')
     ->set('branch', 'staging')
-    ->set('bin/php', 'php8.0')
-    ->set('document_root', '~/php74/{{client}}/{{project}}');
+    ->set('bin/php', '{{php_binary}}')
+    ->set('cachetool', '/var/run/php_{{php_version_numeric}}_fpm_sites.sock')
+    ->set('document_root', '~/php{{php_version_numeric}}/{{client}}/{{project}}');
 
 // Define production
 //host('$host')
-//    ->setRemoteUser(get('production_user'))
+//    ->setRemoteUser('{{production_user}}')
 //    ->set('labels', ['stage' => 'production'])
 //    ->port(2244)
 //    ->set('deploy_path', '~/wwwroot')
 //    ->set('branch', 'master')
-//    ->set('bin/php', 'php8.0')
+//    ->set('bin/php', '{{php_binary}}')
 //    ->set('document_root', '~/wwwroot/www')
-//    ->set('http_user', get('production_user'))
+//    ->set('http_user', '{{production_user}}')
 //    ->set('writable_mode', 'chmod'); Cloudstar only
 
 /*************************
  * No need to edit below *
  *************************/
 
+set('php_binary', function () {
+    return 'php' . get('php_version');
```

Comment:
Hmm, on some systems it is php7_4, others php7.4 I don't know if I agree with building the php_binary dynamically.

Reply:
I wanted to provide a nice default for our own servers, but I can take it out. This way, both the cachecool and doucment_root for staging can be automised:
```php
    ->set('cachetool', '/var/run/php_{{php_version_numeric}}_fpm_sites.sock')
    ->set('document_root', '~/php{{php_version_numeric}}/{{client}}/{{project}}');
    ```
The PHP binary can always be overruled in each host's definition of `bin/php`. But I do agree that dynamically building it is kind of opiniated and will sometimes be wrong.

Reply:
ok, fine by me